`num-traits` `v0.2.0` dependency is too old

`num-traits` `v0.2.0`, which [`fast-hilbert` depends on](https://github.com/becheran/fast-hilbert/blob/c6bd5b18162fe935fe08dd5d36c1d3910815cefe/Cargo.toml#L14), doesn't support `impl PrimInt for u128` but [`fast-hilbert` requires it](https://github.com/becheran/fast-hilbert/blob/c6bd5b18162fe935fe08dd5d36c1d3910815cefe/src/lib.rs#L58-L59).

This caused a compilation error:
```console
| error[E0277]: the trait bound `u128: PrimInt` is not satisfied
|   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/fast_hilbert-1.0.0/src/lib.rs:59:16
|    |
| 59 |     type Key = u128;
|    |                ^^^^ the trait `PrimInt` is not implemented for `u128`
|    |
|    = help: the following other types implement trait `PrimInt`:
|              i16
|              i32
|              i64
|              i8
|              isize
|              u16
|              u32
|              u64
|            and 2 others
| note: required by a bound in `Unsigned::Key`
```

I'll send a PR to fix this! :rocket:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ categories = ["algorithms"]
 repository = "https://github.com/becheran/fast-hilbert"
 
 [dependencies]
-num-traits = "0.2"
+num-traits = "0.2.3"
 
 [dev-dependencies]
 image = "0.23.13"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ categories = ["algorithms"]
 repository = "https://github.com/becheran/fast-hilbert"
 
 [dependencies]
-num-traits = { version = "0.2.4", features = "i128" }
+num-traits = { version = "0.2.4", features = [ "i128" ] }
 
 [dev-dependencies]
 image = "0.23.13"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ categories = ["algorithms"]
 repository = "https://github.com/becheran/fast-hilbert"
 
 [dependencies]
-num-traits = "0.2.3"
+num-traits = "0.2.4"
 
 [dev-dependencies]
 image = "0.23.13"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ categories = ["algorithms"]
 repository = "https://github.com/becheran/fast-hilbert"
 
 [dependencies]
-num-traits = "0.2.4"
+num-traits = { version = "0.2.4", features = "i128" }
 
 [dev-dependencies]
 image = "0.23.13"

